### PR TITLE
update test-infra in vendor

### PR DESCRIPTION
**Pull request by yt3liu**
Part of #https://github.com/knative/test-infra/issues/829

## Proposed Changes

  * update test-infra in vendor

**Release Note**

```release-note
NONE
```

### CODE REVIEW

**Comment by yt3liu** [comment]
pending on removing dev-stat in test-infra /hold

**Comment by adrcunha** [comment]
This PR unnecessarily added the devstats dir under test-infra, please fix.

File: Gopkg.lock [hunk 1]
```diff
@@ -1371,7 +1368,7 @@
   revision = "e494cc58111187acad93e64529228a2fc0153e39"
 
 [[projects]]
-  digest = "1:81d6e6a65b01e29608fc3c5c927bb3c14152e1221b26db46c19694744d2a32a4"
```
Comment (adrcunha) [inline Gopkg.lock]: You're making changes to this entry but not updating any file, is it intended?

Comment (adrcunha) [reply Gopkg.lock]: Looks like not, since build tests failed in verify-codegen.sh.

**Pull request closed without merging**